backend/data/processors/programs_processing.py

```diff
@@ -120,12 +123,24 @@ def addMinorData(components, item):
     components["Minors"] = minorData
 
 
-def addDisciplineData(components, item):
-    # Initialise Specialisation Data (explained in wiki)
-    SpecialisationData = {}
-    # Initialise NonSpecialisation Data (explained in wiki)
-    NonSpecialisationData = {}
+def findProgramName(programData, item):
+    programName = item["title"]
+    programNames = programData["title"].split("/")
+    for programName in programNames:
+        if programName in item["title"]:
+            return programName.strip()
+        for container in item["container"]:
+            if programName in container["title"]:
+                return programName.strip()
+    return programName.strip()
+
+def addDisciplineData(components, item, programData):
+    if components.get("SpecialisationData") is None:
+        components["SpecialisationData"] = {}
+    if components.get("NonSpecialisationData") is None:
+        components["NonSpecialisationData"] = {}
```

Comment:
Sorry I didn't notice this before lol
```suggestion
    components.setdefault("SpecialisationData", {})
    components.setdefault("NonSpecialisationData", {})
```